src/db/pool.cr

```diff
@@ -141,7 +141,9 @@ module DB
       idle_pushed = false
 
       sync do
-        if can_increase_idle_pool
+        if resource.responds_to?(:closed?) && resource.closed?
```

Comment:
I would like to say that is not needed, but some people want to use the pool with http client. The `HTTP::Client` has a `#close`, but has no `#closed?`.

Reply:
We should probably add that to `HTTP::Client` 🤔

Reply:
We could, but the HTTP::Client might be reopened. I'm not sure what a good choice for interacting with it through a pool would be. But for now the `responds_to?` needs to stay because of that.

Reply:
I was wondering if there's an option for more conservative behaviour when the resource does not respond to `#closed?`. But I don't think there is one except unconditionally deletion, which would defeat the purpose of a pool.

Maybe it makes sense to consider a resource type that does not implement `#closed?` as being able to reset/reopen itself on subsequent checkouts.

Reply:
> Maybe it makes sense to consider a resource type that does not implement #closed? as being able to reset/reopen itself on subsequent checkouts.

And that would be the current behavior. Or you think it's not?

Reply:
Yes, that's it.